### ISSUE TITLE
Unhandled 'error' event - write after end

### ISSUE THREAD

**Issue by pworkpop** [grpc/grpc-node#927]
### Problem description
The unhandled error event terminates node process with the following stack trace:

### Reproduction steps
This is a network service using the Google PubSub https://github.com/googleapis/nodejs-pubsub client. Exception is raised presumably due to a network issue that interrupts the connection kept open to the PubSub Streaming Pull endpoint.
The service keeps pulling for queued messages so connection can be open for several days before this error appears.

### Environment
 - OS name, version and architecture: Docker Debian 9 amd64
 - Node version: 8.16.0
 - Node installation method: Docker node:8-stretch-slim
 - If applicable, compiler version
 - Package name and version gRPC@0.4.3, nodejs-pubsub@0.30.0

### Additional context
Obtained the following stack trace:
```
events.js:183
throw er; // Unhandled 'error' event
^

Error: write after end
    at writeAfterEnd (_stream_writable.js:236:12)
    at ClientHttp2Stream.Writable.write (_stream_writable.js:287:5)
    at filterStack.sendMessage.then.message (/usr/src/app/node_modules/@grpc/grpc-js/build/src/call-stream.js:359:34)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:189:7)
```

**Comment by mathieu-jld** [comment]
Any news on this issue ?

**Comment by murgatroid99** [comment]
I'm not sure, but I believe it will be fixed by #1184. That PR includes a major rework of the code that directly throws that error.

**Comment by murgatroid99** [comment]
Those changes have been published in grpc-js version 0.7.x. Please try that out to see if it helps.
